tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py

```diff
@@ -102,3 +105,174 @@ def update_fn(engine, batch):
 
     assert np_median_absolute_relative_error == pytest.approx(median_absolute_relative_error)
+
+
+def _test_distrib_compute(device):
+    rank = idist.get_rank()
+
+    def _test(metric_device):
+        metric_device = torch.device(metric_device)
+        m = MedianRelativeAbsoluteError(device=metric_device)
+        torch.manual_seed(10 + rank)
+
+        size = 151
+
+        y_pred = torch.randint(1, 10, size=(size, 1), dtype=torch.double, device=device)
+        y = torch.randint(1, 10, size=(size, 1), dtype=torch.double, device=device)
+
+        m.update((y_pred, y))
+
+        # gather y_pred, y
+        y_pred = idist.all_gather(y_pred)
+        y = idist.all_gather(y)
+
+        np_y_pred = y_pred.cpu().numpy().ravel()
+        np_y = y.cpu().numpy().ravel()
+
+        res = m.compute()
+
+        e = np.abs(np_y - np_y_pred) / np.abs(np_y - np_y.mean())
+        np_res = np.median(e)
+
+        e_prepend = np.insert(e, 0, e[0], axis=0)
+        np_res_prepend = np.median(e_prepend)
+
+        # The results between numpy.median() and torch.median() are Inconsistant
+        # when the length of the array/tensor is even. So this is a hack to avoid that.
+        # issue: https://github.com/pytorch/pytorch/issues/1837
+        if np_y_pred.shape[0] % 2 == 0:
+            assert pytest.approx(res) == np_res_prepend
+        else:
+            assert pytest.approx(res) == np_res
+
+    for _ in range(3):
+        _test("cpu")
+        if device.type != "xla":
+            _test(idist.device())
+
+
+def _test_distrib_integration(device):
+
+    rank = idist.get_rank()
+    torch.manual_seed(12)
+
+    def _test(n_epochs, metric_device):
+        metric_device = torch.device(metric_device)
+        n_iters = 80
+        size = 151
+        y_true = torch.rand(size=(size,)).to(device)
+        y_preds = torch.rand(size=(size,)).to(device)
+
+        def update(engine, i):
+            return (
+                y_preds[i * size : (i + 1) * size],
+                y_true[i * size : (i + 1) * size],
+            )
+
+        engine = Engine(update)
+
+        m = MedianRelativeAbsoluteError(device=metric_device)
+        m.attach(engine, "mare")
+
+        data = list(range(n_iters))
+        engine.run(data=data, max_epochs=n_epochs)
+
+        assert "mare" in engine.state.metrics
+
+        res = engine.state.metrics["mare"]
+
+        np_y_true = y_true.cpu().numpy().ravel()
+        np_y_preds = y_preds.cpu().numpy().ravel()
+
+        e = np.abs(np_y_true - np_y_preds) / np.abs(np_y_true - np_y_true.mean())
+        np_res = np.median(e)
+
+        e_prepend = np.insert(e, 0, e[0], axis=0)
+        np_res_prepend = np.median(e_prepend)
+
+        # The results between numpy.median() and torch.median() are Inconsistant
+        # when the length of the array/tensor is even. So this is a hack to avoid that.
+        # issue: https://github.com/pytorch/pytorch/issues/1837
+        if np_y_preds.shape[0] % 2 == 0:
```

Comment:
Total size is 151 here, I think we do not need this trick here ?

Reply:
Yes we are not gathering here, thanks for this!